crates/api/src/widget/state.rs

```diff
@@ -7,23 +7,80 @@ pub trait AsAny: Any {
     fn as_any_mut(&mut self) -> &mut dyn Any;
 }
 
-/// Used to define a state of a widget.
+/// Used to define a state of a [`widget`].
+/// 
+/// The state holds the logic of a widget which makes it interactive.
+/// The state of a widget made up of a struct which implements this trait with its fields and its methods.
+/// A state of a widget is represented by the current values of its properties.
+/// Each state has to implement this trait.
+/// Each state has to derive or implement the [Default](https://doc.rust-lang.org/std/default/trait.Default.html) and the [`AsAny`] traits.
+/// A state is operating on the properties (components) of the widget, its parent or children, or the state's fields.
+/// It is not mandatory to have a state for a widget (in this case it will be static).
+/// 
+/// # Example
+/// ```
+/// use orbtk::prelude::*;
+/// 
+/// #[derive(Default, AsAny)]
+/// struct MyState {
+///     count: usize
+/// }
+/// 
+/// impl State for MyState {
+///     fn init(&mut self, _registry: &mut Registry, _ctx: &mut Context) {
+///         self.count = 42;
+///         println!("MyState initialized.");
+///     }
+/// 
+///     fn update(&mut self, _registry: &mut Registry, _ctx: &mut Context) {
+///         self.count += 1;
+///         println("MyState updated.");
+///     }
 ///
-/// A state is used to operate on the properties (components) of the widget, its parent or children.
+///     fn update_post_layout(&mut self, _registry: &mut Registry, _ctx: &mut Context) {
+///         println("MyState updated after layout is calculated.");
+///     }
+/// }
+/// 
+/// widget!(MyWidget<MyState>)
+/// ```
+/// 
+/// [`widget`]: ./trait.Widget.html
+/// [`AsAny`]: ./trait.AsAny.html
+
 pub trait State: AsAny {
-    /// Init is used for initial setup.
-    fn init(&mut self, _: &mut Registry, _: &mut Context) {}
+    /// Init is used for setup the initial state of a widget, setting fields to starting values, registering service(s).
+    /// It is called after the widget is created.
+    /// 
+    /// # Arguments
+    /// * `_registry`: Gives you access to the global Service Registry.
```

Comment:
I don't think using "you" is appropriate for documentation.  Should be perspective agnostic.  So:
Provides access ...